apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -202,14 +220,18 @@
         }
 
         this.states.passCodeEntered += key;
-        this.updatePassCodeUI();
+        this.updatePassCodeUI('');
 
         if (this.states.padVibrationEnabled) {
           navigator.vibrate(this.configs.padVibrationDuration);
         }
 
         if (this.states.passCodeEntered.length === 4) {
-          this.lockScreen.checkPassCode(this.states.passCodeEntered);
+          (function(value) {
+            setTimeout(function() {
+              this.lockScreen.checkPassCode(value);
+            }, CODE_VALIDATION_TIMEOUT);
```

Comment:
It seems like terrible UX to make the user wait an extra second after entering their pin before unlocking.  That will be infuriating. And if the UX spec says to do that, then we should get the spec created. UX designers are not gods... they make mistakes too, and when they tell us to do something stupid, we should let them know!  :-)